### PR TITLE
feat: Added Global rate limiting middleware - Implemented RateLimiter…

### PR DESCRIPTION
**Pull request by RishabhC-137**
This PR implements the feature described in [DiceDB issue #729](https://github.com/DiceDB/dice/issues/729).
This PR:
-Implements a global rate limiter for the /command endpoint, allowing up to 1000 requests per minute.
-Utilizes DiceDB commands (GET, INCR, and EXPIRE) to track and manage request counts.
-Blocks requests exceeding the limit and returns a 429 Too Many Requests response.
-Ensures the request count resets automatically at the start of each new minute.
-Excludes the /search endpoint from rate limiting.

### CODE REVIEW

**Comment by lucifercr07** [comment]
@RishabhC-137 can you please close and reopen the PR, this would allow the workflows to trigger and validate.

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	RedisAddr     string
+	ServerPort    string
+	RequestLimit  int // Field for the request limit
+	RequestWindow int // Field for the time window in seconds
+}
+
+// LoadConfig loads the application configuration from environment variables or defaults
+func LoadConfig() *Config {
+	return &Config{
+		RedisAddr:     getEnv("REDIS_ADDR", "localhost:7379"), // Default Redis address
```
Comment (lucifercr07) [inline config/config.go:19]: This is not required, we'd be using DiceDB.

File: main.go [hunk 2]
```diff
@@ -4,25 +4,45 @@ import (
...
+	redis "github.com/dicedb/go-dice"
 )
 
 type HTTPServer struct {
 	httpServer *http.Server
+	diceClient *redis.Client
 }
 
-func NewHTTPServer(addr string, mux *http.ServeMux) *HTTPServer {
+func NewHTTPServer(addr string, mux *http.ServeMux, client *redis.Client) *HTTPServer {
```
Comment (lucifercr07) [inline main.go:22]: Let not use redis wording, we can change this to `*dice.Client`

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (RishabhC-137) [reply main.go]: I have made the changes to replace all instances of 'Redis' with 'dice.Client' throughout the code and documentation. Everything is now updated accordingly.

File: config/config.go (hunk 1 above)
Comment (RishabhC-137) [reply config/config.go]: Made the Changes

File: internal/middleware/ratelimiter.go [hunk 3]
```diff
@@ -1,11 +1,83 @@
...
 
-func RateLimiter(next http.Handler) http.Handler {
+// RateLimiter middleware to limit requests based on a specified limit and duration
+func RateLimiter(diceClient *dice.Client, next http.Handler, limit, window int) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		// Check DiceDB connection health
+		if err := diceClient.Ping(ctx).Err(); err != nil {
```
Comment (lucifercr07) [inline internal/middleware/ratelimiter.go:21]: We don't need to ping DiceDB for dice health, anyways `Get` would send and err if DB is down, we can avoid this extra query to DB.

**Review by lucifercr07 — APPROVED** [review]
LGTM, for initial implementation. We can iterate on this further.
I'm thinking that since APIs would be public there'd be potential abuses to the system.
@RishabhC-137 thanks for contributing.

Have a minor comment please have a look.

**Pull request merged**